CalendarApp/NUnitTestProject/BookingServiceTests.UpdateTest.cs

```diff
@@ -0,0 +1,98 @@
+                throw new Exception("Event list are empty!");
+            var testEvent = beforeUpdateEventList[0];
+
+            var currentSummary = testEvent.Summary; //remember current state
+            var currentDesctiption = testEvent.Description;
+            testEvent.Summary = "ChangedSummary"; //chenage event
+            testEvent.Description = "ChengedDescription";
+
+            _meetingBooking.UpdateEvent(testEvent, calendarId, testEvent.Id);
+
+            var ufterApdateList = _calendarService.Events.List(calendarId).Execute().Items;
+
+            var testedAfterUpdateEvent = (from gEvent in ufterApdateList
+                                          where gEvent.Id == testEvent.Id
+                                          select gEvent).First();
+            if (testedAfterUpdateEvent == null)
+                throw new Exception("Cant Find updated Event");
+
+            Assert.AreEqual(testEvent.Summary, testedAfterUpdateEvent.Summary); //we cant just Assert.AreEqual(two this object) couse of "updated" time field will be different anyway
+            Assert.AreEqual(testEvent.Description, testedAfterUpdateEvent.Description);
+
+            testedAfterUpdateEvent.Summary = currentSummary;//push event back to past state
+            testedAfterUpdateEvent.Description = currentDesctiption;
+
+            _meetingBooking.UpdateEvent(testedAfterUpdateEvent, calendarId, testedAfterUpdateEvent.Id);
+
+            var checkBackEvent = (from gEvent in beforeUpdateEventList
+                                  where gEvent.Id == testEvent.Id
+                                  select gEvent).Single();
+            if (checkBackEvent == null)
+                throw new Exception("Cant Find updated Event");
+
+            Assert.AreEqual(checkBackEvent.Summary, testEvent.Summary);
+            Assert.AreEqual(checkBackEvent.Description, testEvent.Description);
+        }
+
+        [Test]
+        public void BookingService_UpdateEvent_WrongCalendarIdThrowException()
+        {
+            var calendarId = AppSettingsHelper.GetAppSetting(AppSetingsConst.TestCalendar);
+
+            var beforeUpdateEventList = _calendarService.Events.List(calendarId).Execute().Items;
+            if (beforeUpdateEventList == null)
+                throw new Exception("Empty event list");
+            var testEvent = beforeUpdateEventList[0];//take any event
+
+            Assert.Throws(typeof(Google.GoogleApiException),
+                () => { _meetingBooking.UpdateEvent(testEvent, string.Empty, testEvent.Id); });
+        }
+
+        [Test]
+        public void BookingService_UpdateEvent_WrongEventIdThrowException()
+        {
+            var calendarId = AppSettingsHelper.GetAppSetting(AppSetingsConst.TestCalendar);
+
+            var beforeUpdateEventList = _calendarService.Events.List(calendarId).Execute().Items;
+            if (beforeUpdateEventList == null)
+                throw new Exception("Empty event list");
```

Comment:
Still curiously why has exception been thrown here (and in another occurred places)?